sycl/source/detail/device_image_impl.hpp

```diff
@@ -278,20 +278,38 @@ class device_image_impl {
                                                             Descriptors.size());
         unsigned LocalOffset = 0;
         while (It != End) {
-          // Make sure that alignment is correct in blob.
-          const unsigned OffsetFromLast = /*Offset*/ It[1] - LocalOffset;
-          BlobOffset += OffsetFromLast;
-          // Composites may have a special padding element at the end which
-          // should not have a descriptor. These padding elements all have max
-          // ID value.
-          if (It[0] != std::numeric_limits<std::uint32_t>::max()) {
+          if (MBinImage->supportsSpecConstants()) {
+            // Make sure that alignment is correct in blob.
+            const unsigned OffsetFromLast = /*Offset*/ It[1] - LocalOffset;
+            BlobOffset += OffsetFromLast;
+
+            // Composites may have a special padding element at the end which
+            // should not have a descriptor. These padding elements all have max
+            // ID value.
+            if (It[0] != std::numeric_limits<std::uint32_t>::max()) {
+              // The map is not locked here because updateSpecConstSymMap() is
+              // only supposed to be called from c'tor.
+              MSpecConstSymMap[std::string{SCName}].push_back(
```

Comment:
This call can be moved out of the if-then-else (nit pick). Thanks

Reply:
Extracted it and also re-worked this part of the code so it's more readable, mainly re-named the iterators.